bundle/src/main/java/com/adobe/acs/commons/users/impl/EnsureAce.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * #%L ACS AEM Commons Bundle %% Copyright (C) 2017 Adobe %% Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License. #L%
+ */
+
+package com.adobe.acs.commons.users.impl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jcr.PropertyType;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.jcr.Value;
+import javax.jcr.ValueFactory;
+import javax.jcr.ValueFormatException;
+import javax.jcr.security.AccessControlPolicy;
+import javax.jcr.security.Privilege;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlEntry;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlList;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlManager;
+import org.apache.jackrabbit.api.security.user.Authorizable;
+import org.apache.jackrabbit.commons.jackrabbit.authorization.AccessControlUtils;
+import org.apache.jackrabbit.oak.spi.security.authorization.accesscontrol.AccessControlConstants;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.day.cq.search.PredicateGroup;
+import com.day.cq.search.Query;
+import com.day.cq.search.QueryBuilder;
+
+@Component
+@Service(EnsureAce.class)
+public class EnsureAce {
+
+    private static final Logger log = LoggerFactory.getLogger(EnsureAce.class);
+
+    @Reference
+    private QueryBuilder queryBuilder;
+
+    /**
+     * Ensures the ACEs for the Service User exists. Any extra ACEs for the Service User will be removed.
+     *
+     * @param resourceResolver
+     *            the resource resolver to perform the user management
+     * @param authorizable
+     *            the Authorizable (User or Group) the ServiceUser represents
+     * @param serviceUser
+     *            the Service User return # of ace entries that could not be processed
+     * @throws RepositoryException
+     */
+    @SuppressWarnings("squid:S3776")
+    public int ensureAces(ResourceResolver resourceResolver, Authorizable authorizable,
+            AbstractAuthorizable serviceUser) throws RepositoryException {
```

Comment:
WDYT about renaming this variable from serviceUser -> authorizable (and its cousins in the methods) ? When i read through it the first time, it made it sound like this was explicitly for serviceUsers because all the vars include "serviceUser".